browser/core_metrics/core_metrics_service_unittest.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+
+#include "base/test/metrics/histogram_tester.h"
+#include "brave/components/core_metrics/core_metrics_service.h"
+#include "chrome/browser/history/history_service_factory.h"
+#include "chrome/test/base/testing_profile_manager.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/keyed_service/core/service_access_type.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace core_metrics {
+
+class CoreMetricsServiceUnitTest : public testing::Test {
+ public:
+  CoreMetricsServiceUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+
+  void SetUp() override {
+    TestingProfile::Builder builder;
+    builder.AddTestingFactory(HistoryServiceFactory::GetInstance(),
+                              HistoryServiceFactory::GetDefaultFactory());
+    profile_ = builder.Build();
+
+    history_service_ = HistoryServiceFactory::GetForProfile(
+        profile_.get(), ServiceAccessType::EXPLICIT_ACCESS);
+    core_metrics::CoreMetricsService::RegisterPrefs(local_state_.registry());
+    core_metrics_service_ =
+        new CoreMetricsService(&local_state_, history_service_);
+  }
+
+ protected:
+  content::BrowserTaskEnvironment task_environment_;
+  TestingPrefServiceSimple local_state_;
+  base::HistogramTester histogram_tester_;
+  std::unique_ptr<TestingProfile> profile_;
+  raw_ptr<history::HistoryService> history_service_;
+  raw_ptr<CoreMetricsService> core_metrics_service_;
+};
+
+TEST_F(CoreMetricsServiceUnitTest, DomainsLoadedCount) {
+  histogram_tester_.ExpectTotalCount(kDomainsLoadedHistogramName, 0);
+
+  task_environment_.FastForwardBy(base::Seconds(30));
+
+  histogram_tester_.ExpectUniqueSample(kDomainsLoadedHistogramName, 0, 1);
+
+  history_service_->AddPage(GURL("https://abc.com"), base::Time::Now(),
+                            history::VisitSource::SOURCE_BROWSED);
+  history_service_->AddPage(GURL("https://def.org"), base::Time::Now(),
+                            history::VisitSource::SOURCE_BROWSED);
+  history_service_->AddPage(GURL("https://xyz.org"), base::Time::Now(),
+                            history::VisitSource::SOURCE_BROWSED);
+  history_service_->AddPage(GURL("https://xyz.net/page1"), base::Time::Now(),
+                            history::VisitSource::SOURCE_BROWSED);
+  history_service_->AddPage(GURL("https://xyz.net/page2"), base::Time::Now(),
+                            history::VisitSource::SOURCE_BROWSED);
+
+  histogram_tester_.ExpectBucketCount(kDomainsLoadedHistogramName, 1, 0);
```

Comment:
I suggest replacing `ExpectBucketCount` to `ExpectUniqueSample` wherever it possible ;)

Reply:
I think the first sample is the only case that can be checked with `ExpectUniqueSample`